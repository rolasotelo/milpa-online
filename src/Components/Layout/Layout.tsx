import React, { ReactNode } from "react";
import { useHistory } from "react-router-dom";
import logo from "./../../../static/logo.png";

interface Props {
  children: ReactNode;
}

const Layout = (props: Props) => {
  const h = useHistory();
  return (
    <div className="bg-mexicanBone min-h-screen">
      <div className="bg-mexicanBlue">
        <div className="container mx-auto p-2">
          <nav className="flex justify-between text-mexicanPink">
            <div className="self-center">
              <button
                className="bg-button-menu-up focus:bg-button-menu-down focus:outline-none"
                style={{
                  width: "50px",
                  height: "50px",
                }}
              ></button>
            </div>
            <div>
              <button
                onClick={() => {
                  h.push("/");
                }}
                className="bg-logo-floating focus:outline-none"
                style={{
                  width: "50px",
                  height: "50px",
                }}
              ></button>
            </div>
            <div className="self-center">
              <button
                className="bg-button-leaderboard-up focus:bg-button-leaderboard-down focus:outline-none"
                style={{
                  width: "50px",
                  height: "50px",
                }}
              ></button>
            </div>
          </nav>
        </div>
      </div>
      <div className="max-w-6xl flex flex-col  py-5 mx-auto">
        {props.children}
      </div>
    </div>
  );
};

export default Layout;
