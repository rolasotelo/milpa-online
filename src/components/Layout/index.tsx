import React, { ReactNode } from "react";
import Navbar from "../Navbar";

interface Props {
  children: ReactNode;
}

function Layout(props: Props) {
  const { children } = props;
  return (
    <div className="bg-milpaBeige-default min-h-screen overflow-x-hidden">
      <Navbar />
      <div className="flex flex-col mx-auto">{children}</div>
    </div>
  );
}

export default Layout;
