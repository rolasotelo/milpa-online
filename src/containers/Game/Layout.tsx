import React, {ReactElement, ReactNode} from "react";

interface PropsWithChildren {
    children: ReactNode;
}

export default function Layout(props: PropsWithChildren): ReactElement {
    const { children } = props;
    return (
        <div className="relative flex-1 flex flex-row-reverse tablet:flex-col w-full pr-20 tablet:pr-0">
            {children}
        </div>
    );
}