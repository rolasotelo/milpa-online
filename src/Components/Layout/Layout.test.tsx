import React from "react";
import renderer from "react-test-renderer";

import Layout from "./Layout";

describe("<Layout />", () => {
  const defaultProps = {
    children: <div>Hola</div>,
  };
  const wrapper = renderer.create(<Layout {...defaultProps} />);

  test("render", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
