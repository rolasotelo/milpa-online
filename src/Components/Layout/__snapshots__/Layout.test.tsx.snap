// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> render 1`] = `
<div
  className="bg-mexicanBone min-h-screen"
>
  <div
    className="bg-mexicanBlue"
  >
    <div
      className="container mx-auto p-2"
    >
      <nav
        className="flex justify-between text-mexicanPink"
      >
        <div
          className="self-center"
        >
          <button
            className="bg-button-menu focus:bg-button-menu-pressed focus:outline-none"
            style={
              Object {
                "height": "50px",
                "width": "50px",
              }
            }
          />
        </div>
        <div>
          <button
            className="bg-button-logo focus:bg-button-logo-pressed focus:outline-none"
            onClick={[Function]}
            style={
              Object {
                "height": "50px",
                "width": "50px",
              }
            }
          />
        </div>
        <div
          className="self-center"
        >
          <button
            className="bg-button-leaderboard focus:bg-button-leaderboard-pressed focus:outline-none"
            style={
              Object {
                "height": "50px",
                "width": "50px",
              }
            }
          />
        </div>
      </nav>
    </div>
  </div>
  <div
    className="max-w-6xl flex flex-col  py-2 mx-auto"
  >
    <div>
      Hola
    </div>
  </div>
</div>
`;
